icecreamshop/icecreamshop/src/main/java/com/summerschool/icecreamshop/service/ProductService.java

```diff
@@ -21,4 +21,24 @@ public List<Product> getAll() {
         return productRepository.findAll();
     }
 
+    public void add(Product product){
+        productRepository.save(product);
+    }
+
+    public boolean isProductValid(Product product){
+        if (product.getTitle().isEmpty()
+            || product.getShortDescription().isEmpty()
+            || product.getLongDescription().isEmpty()
+            || product.getIngredients().isEmpty()
+            || product.getQuantity() != 0
+            || product.getAlergens().isEmpty()
+            || product.getPrice() != 0
+            || product.getCurrency().isEmpty()
+            || product.getPhotoUrl().isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
```

Comment:
You should use the @Valid annotation instead of having this method to check if the Product is valid.

Reply:
@teodorasteluta poti rezolva acest comentariu te rog?